app/src/highlighter/index.ts

```diff
@@ -152,8 +155,9 @@ function guessMimeType(contents: string) {
 
 function detectMode(request: IHighlightRequest): CodeMirror.Mode<{}> | null {
   const mimeType =
-    extensionMIMEMap.get(request.extension.toLowerCase()) ||
-    guessMimeType(request.contents)
+    extensionMIMEMap.get(
+      request.extension.toLowerCase() || request.basename.toLowerCase()
+    ) || guessMimeType(request.contents)
```

Comment:
I kinda feel like thats wrong as Dockerfile in particular is not supposed to have an extension

Reply:
Think about Dockerfile.java. Is it Java or Docker?

Reply:
I think that we might be overloading the `extensionMIMEMap` here and that's making it tricky to reason about. Since this is introducing a special case for files which have an exact basename match (seeing as a basename includes the extension if one exists) I think we should have a separate map for that. How about we create a `basenameMIMEMap` and check that first before checking the `extensionMIMEMap`?